scheduler/cloudformation/cloudformation.go

```diff
@@ -416,41 +454,23 @@ func (s *Scheduler) waitUntilStackOperationComplete(lock *pglock.AdvisoryLock, w
 }
 
 // executeStackUpdate performs a stack update.
-func (s *Scheduler) executeStackUpdate(input *cloudformation.UpdateStackInput) error {
+func (s *Scheduler) executeStackUpdate(input *updateStackInput) error {
 	stack, err := s.stack(input.StackName)
 	if err != nil {
 		return err
 	}
 
-	// If we're updating a stack, without changing the template, merge in
-	// existing parameters with their previous value.
-	if input.UsePreviousTemplate != nil && *input.UsePreviousTemplate == true {
-		// The parameters that the stack defines. We need to make sure that we
-		// provide all parameters in the update (lame).
-		definedParams := make(map[string]bool)
-		for _, p := range stack.Parameters {
-			definedParams[*p.ParameterKey] = true
-		}
-
-		// The parameters that are provided in this update.
-		providedParams := make(map[string]bool)
-		for _, p := range input.Parameters {
-			providedParams[*p.ParameterKey] = true
-		}
-
-		// Fill in any parameters that weren't provided with their default
-		// value.
-		for k := range definedParams {
-			if !providedParams[k] {
-				input.Parameters = append(input.Parameters, &cloudformation.Parameter{
-					ParameterKey:     aws.String(k),
-					UsePreviousValue: aws.Bool(true),
-				})
-			}
-		}
+	i := &cloudformation.UpdateStackInput{
+		StackName:  input.StackName,
+		Parameters: updateParameters(input.Parameters, stack, input.Template),
+	}
+	if input.Template != nil {
```

Comment:
seems like: https://github.com/remind101/empire/pull/898/files#diff-67e3f5de813e6b6e24cb66d3278cff99R208 will only return a `nil` template on error, otherwise you should always have this?